### DISCUSSION TITLE
How can I get an STM32F103C8 running at 72MHz with ArduinoIDE and HAL library?

### DISCUSSION THREAD

**Discussion by Frankie6626** [stm32duino/Arduino_Core_STM32#2023] [Q&A] [answered]
I've posted at StackOverflow and StackExchange, but no one knows better than people from here. The original question is in the link below. 
https://stackoverflow.com/questions/76318971/how-can-i-get-an-stm32f103c8-running-at-72mhz-with-arduinoide-and-hal-library
We got it working with STM32CubeIDE at 72 MHz. After we copied it to ArduinoIDE with STM32duino implemented, the chip ran at 36 MHz. Someone asked how RCC registers are set up, but I need help determining which file is used and where it is defined. This one might help to determine where HSE is defined and how the entire clock is set up.

**Comment by fpistm** [comment, marked as answer]
Hi @Frankie6626 
You can easily redefine the `SystemClock_Config` at sketch level as stated in the wiki:
https://github.com/stm32duino/Arduino_Core_STM32/wiki/Custom-definitions#systemclock_config

But you have to pay attention that by default HSE_VALUE for STM32F1xx serie is 8MHz:
https://github.com/stm32duino/Arduino_Core_STM32/blob/052a49af1615fb5468ef603a0213d490d810afa2/system/STM32F1xx/stm32f1xx_hal_conf_default.h#L86

So you have to redefine it globally to 16MHz, that's why you do not have the correct freq (half), to do that you can use build_opt.h;
https://github.com/stm32duino/Arduino_Core_STM32/wiki/Customize-build-options-using-build_opt.h

with `-DHSE_VALUE=16000000U `
Then you should have correct freq.

**Reply by Frankie6626** [reply]
Thank you very much. This is exactly what I need.
